### PR TITLE
Extend ON DUPLICATE KEY UPDATE support

### LINKED ISSUE #1817: About MySQLQueryFactory.insertOnDuplicateKeyUpdate.
I'm trying to use  MySQLQueryFactory.insertOnDuplicateKeyUpdate.
but I didn't find setting to specific columns for insert and there isn't multiple update clause.
I hope these feature will be added.

sample sql
```
insert into content_count_day_log
		(
		 day_format
		, object_id
		, content_count
		, mod_datetime
		)
		select
				substr(hour_format,1,8)  day_format
				, object_id
				, sum(content_count) content_count
				, now()
			from content_count_hour_log l
			where hour_format >= #{fromDate}
				and hour_format < #{toDate}
			group by
				day_format
				, object_id
		on duplicate key update
		content_count = values(content_count)						
		, mod_datetime = now()
```

sample querydsl
```
this.queryService
                .insertOnDuplicateKeyUpdate(this.qContentCountDayLog,
                    this.qContentCountDayLog.contentCount.eq(qContentCountHourLog.contentCount))
                .select(
                    this.queryService
                        .select(
                            qContentCountHourLog.hourFormat.substring(0, 8).as("dayFormat"),
                            qContentCountHourLog.objectId,
                            qContentCountHourLog.contentCount.sum().as("contentCount")
                        )
                        .from(qContentCountHourLog)
                        .where(qContentCountHourLog.hourFormat.goe("2016110100"))
                        .where(qContentCountHourLog.hourFormat.loe("2016110101"))
                        .groupBy(
                            qContentCountHourLog.hourFormat.substring(0, 8),
                            qContentCountHourLog.objectId
                        )
                )
                .execute();
```

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1817

### CODE REVIEW

File: querydsl-sql/src/test/java/com/querydsl/sql/mysql/MySQLQueryFactoryTest.java [hunk 1]
```diff
@@ -82,6 +82,14 @@ public void insertOnDuplicateKeyUpdate2() {
     }
 
     @Test
+    public void insertOnDuplicateKeyUpdate_multiple() {
+        SQLInsertClause clause = queryFactory.insertOnDuplicateKeyUpdate(QSurvey.survey,
+                QSurvey.survey.id.eq(2), QSurvey.survey.name.eq("B"));
+        assertEquals("insert into SURVEY\nvalues () on duplicate key update SURVEY.ID = ?, SURVEY.NAME = ?", clause.toString());
```
Comment (uuidcode) [inline querydsl-sql/src/test/java/com/querydsl/sql/mysql/MySQLQueryFactoryTest.java]: is it possible to write values expression like below sql?
```
on duplicate key update
content_count = values(content_count)
```

Comment (timowest) [inline querydsl-sql/src/test/java/com/querydsl/sql/mysql/MySQLQueryFactoryTest.java]: values is not yet supported, but I can add support in this PR

Comment (uuidcode) [inline querydsl-sql/src/test/java/com/querydsl/sql/mysql/MySQLQueryFactoryTest.java]: Wow. It's good news. thanks, timowest.

**Comment by Shredder121** [comment]
Restarted the stalled builds.

**Comment by Shredder121** [comment]
Looks fine to me by the way.

**Pull request merged**